src/openforms/appointments/contrib/jcc/plugin.py

```diff
@@ -0,0 +1,183 @@
+                appDuration=0,
+            )
+            return days
+        except Exception as e:
+            logger.exception(e)
+            return []
+
+    def get_times(
+        self, products: list, location: AppointmentLocation, day: date
+    ) -> list:
+        product_ids = squash_ids(products)
+
+        try:
+            times = self.client.service.getGovAvailableTimesPerDay(
+                date=day,
+                productID=product_ids,
+                locationID=location.identifier,
+                appDuration=0,
+            )
+            return times
+        except Exception as e:
+            logger.exception(e)
+            return []
+
+    def create_appointment(
+        self,
+        products: list,
+        location: AppointmentLocation,
+        start_at: datetime,
+        client: AppointmentClient,
+        remarks: str = None,
+    ) -> str:
+        product_ids = squash_ids(products)
+
+        try:
+            factory = self.client.type_factory("http://www.genericCBS.org/GenericCBS/")
+            appointment_details = factory.AppointmentDetailsType(
+                locationID=location.identifier,
+                productID=product_ids,
+                clientLastName=client.last_name,
+                clientDateOfBirth=client.birthdate,
+                appStartTime=start_at,
+                appEndTime=start_at,  # Needed but unused by the service.
+                isClientVerified=False,
+                # Optional fields.
+                # These might be needed. Depends on `GetRequiredClientFields`
+                #
+                # clientID=bsn,
+                # clientSex="M/F",
+                # clientInitials="",
+                # clientAddress="",
+                # clientPostalCode="",
+                # clientCity="",
+                # clientCountry="",
+                # clientTel="",
+                # clientMail="",
+                appointmentDesc=remarks,
+                # caseID": "",
+            )
+
+            result = self.client.service.bookGovAppointment(
+                appDetail=appointment_details, fields=[]
+            )
+
+            if result["updateStatus"] == 0:
+                return result["appID"]
+            else:
+                raise Exception(
+                    "Could not create appointment (updateStatus=%s)",
+                    result["updateStatus"],
+                )
+
+        except Exception as e:
+            logger.exception(e)
+            raise
+
+    def delete_appointment(self, identifier: str) -> bool:
+        try:
+            result = self.client.service.deleteGovAppointment(appID=identifier)
+
+            if result == 0:
+                return True
+            else:
+                raise Exception(
+                    "Could not delete appointment (updateStatus=%s)",
+                    result,
+                )
+
+        except Exception as e:
+            logger.exception(e)
+            return False
```

Comment:
See earlier typehint remarks & this would really benefit from the proposed exception-flow rather than returning booleans.

Reply:
Done